lib/ex_phone_number/metadata/phone_metadata.ex

```diff
@@ -43,41 +42,74 @@ defmodule ExPhoneNumber.Metadata.PhoneMetadata do
-        available_formats: [
-          ~x"./availableFormats/numberFormat"el,
-          number_format: ~x"."e |> transform_by(&NumberFormat.from_xpath_node/1)
-        ]
-      )
+      xpath_node
+      |> xmap(
+           id: ~x"./@id"s,
+           country_code: ~x"./@countryCode"i,
+           leading_digits: ~x"./@leadingDigits"o |> transform_by(&normalize_pattern/1),
+           international_prefix: ~x"./@internationalPrefix"s,
+           preferred_international_prefix: ~x"./@preferredInternationalPrefix"o |> transform_by(&normalize_string/1),
+           national_prefix: ~x"./@nationalPrefix"s,
+           national_prefix_for_parsing: ~x"./@nationalPrefixForParsing"o |> transform_by(&normalize_string/1),
+           national_prefix_transform_rule: ~x"./@nationalPrefixTransformRule"o |> transform_by(&normalize_rule/1),
+           national_prefix_formatting_rule: ~x"./@nationalPrefixFormattingRule"s,
+           preferred_extn_prefix: ~x"./@preferredExtnPrefix"o |> transform_by(&normalize_string/1),
+           main_country_for_code: ~x"./@mainCountryForCode"o |> transform_by(&normalize_boolean/1),
+           mobile_number_portable_region: ~x"./@mobileNumberPortableRegion"o |> transform_by(&normalize_boolean/1),
+           carrier_code_formatting_rule: ~x"./@carrierCodeFormattingRule"s,
+           general: ~x"./generalDesc"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           fixed_line: ~x"./fixedLine"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           mobile: ~x"./mobile"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           toll_free: ~x"./tollFree"o |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           premium_rate: ~x"./premiumRate"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           shared_cost: ~x"./sharedCost"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           personal_number: ~x"./personalNumber"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           voip: ~x"./voip"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           pager: ~x"./pager"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           uan: ~x"./uan"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           voicemail: ~x"./voicemail"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           no_international_dialing: ~x"./noInternationalDialling"e |> transform_by(&PhoneNumberDescription.from_xpath_node/1),
+           available_formats: [
+             ~x"./availableFormats/numberFormat"el,
+             number_format: ~x"."e |> transform_by(&NumberFormat.from_xpath_node/1)
+           ]
+         )
+
+    kwlist = if(is_map(kwlist) && is_map(kwlist.general)) do
+      set_general_possible_lengths(kwlist)
+    else
+      kwlist
+    end
+
     struct(%PhoneMetadata{}, kwlist)
   end
 
+  defp set_general_possible_lengths(kwlist) do
```

Comment:
Maybe rewrite as:
```elixir
[
  :fixed_line,
  :mobile,
  ...
]
|> Enum.flat_map(fn type ->
	kwlist
    |> Map.get(type)
    |> get_possible_length()
end)
```

Reply:
Not possible - if the result of it is `nil`, it throws an error `protocol Enumerable not implemented for nil`. I changed this to `Enum.map` and left the `List.flatten` after, then it works.